src/services/download/mod.rs

```diff
@@ -128,15 +130,16 @@ impl DownloadService {
         hub: Arc<Hub>,
     ) -> impl Future<Output = Result<Vec<SourceFileId>, DownloadError>> {
         let worker = self.worker.clone();
+        let config = self.config.clone();
+
         async move {
             match source {
                 SourceConfig::Sentry(cfg) => {
+                    let job =
+                        move || sentry::list_files(cfg, filetypes, object_id, config).bind_hub(hub);
```

Comment:
Does this always clone or is the compiler smart enough to only clone for the sentry source?

Reply:
This likely always clones, I think I would have to write two match stmts to effectively clone only once.

Reply:
I don't thikn I can do this, it makes my future non-static either way